### DISCUSSION TITLE
How to easily get FastAPI system information inside of a function ?

### DISCUSSION THREAD

**Discussion by legalla** [fastapi/fastapi#6997] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Description

Hi all,
I would like to implement a kind of "status" API, which would return different informations about the system current status.
Within the response, I would like, as an example, to get some specific keys, like : 
- FastAPI version
- OpenAPI version
- Redoc version
- Uvicorn version
- Starlette and Pydantic version
- My APIRouter version

Is there an easy way to get those information ?
OpenAPI and APIRouter versions are included in the openapi.json doc, but how can trigger it easily within a function ?
for FastAPI and Uvicorn versions, I assume I should trigger a shell within my code, except if someone as an easy way to do so...

Thanks again,

### Environment

* OS: [e.g. Linux / Windows / macOS]: Docker with Tiangolo version of FastAPI
* FastAPI Version [e.g. 0.3.0]: 0.62.0
* Python version: Python 3.7.7

**Comment by insomnes** [comment, marked as answer]
@legalla for python packages you can get version like this:
Python >= 3.8
```
from importlib.metadata import version
version('pydantic')
```
Python < 3.8
```
import pkg_resources
pkg_resources.get_distribution('pydantic').version
```
The string that you pass to the get_distribution method should be the package name as registered in PyPI. As i know for FastAPI, Pydantic, Starlette and Uvicorn they are the same.

> OpenAPI and APIRouter versions are included in the openapi.json doc, but how can trigger it easily within a function ?

You can use `app.openapi()` this will return openapi as dict

**Comment by legalla** [comment]
Hi @insomnes,

Thanks a lot for your answer !
For the pkg_resources, that's what I've found just before seeing your answer.
So thanks a lot for the confirmation :-)

And thanks a lot for the access to the openapi.json access !
I'll give it a try.

**Comment by Kludex** [comment]
@legalla Do you mind closing the issue (if your problem was solved)? :kissing:

**Comment by legalla** [comment]
Ok, it works as expected :-)
Thanks again, @insomnes

**Comment by tiangolo** [comment]
Thanks for the help here @insomnes and @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @legalla  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
